mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -160,7 +160,7 @@ data class MPSNode(val node: SNode) : IDeprecatedNodeDefaults {
 
     override fun getPropertyValue(property: IProperty): String? {
         if (property.isIdProperty()) {
-            return node.nodeId.toString()
+            return node.reference.toString()
```

Comment:
If that effectively changes the content of the #mpsNodeId# property, we should probably rename it and have to mark this as a breaking change.

Reply:
We also stored the reference in the past. This is a bug, that was introduced in September of last year.
Technically it is a breaking change, but also not really.
I'm not sure if we should declare it breaking.

Reply:
Our client code at least expects a node id there.

Reply:
We could add a second property called "node reference" and put the full reference there. That would also be more consistent.

Reply:
That would add more redundancy though. Since the reference ends with the nodeId.
If a user needs the node id they can parse it from the reference.

Reply:
The bulk-sync should store `INode.reference.serialize()` and the code for that belongs into the synchronizer and not into `MPSNode`. `MPSNode` doesn't need to know anything about that special property.

`#mpsNodeId#` is not a suitable name for that value. It was originally used by the legacy-sync-plugin. I already changed it to store the reference into a property named `$orignalId` (see https://github.com/modelix/modelix.mps-plugins/blob/d9bcc94cfe0bcbc02a0b61ffb88168d33b6502cd/mps-legacy-sync-plugin/src/main/java/org/modelix/model/mpsplugin/MPSNodeMapping.kt#L12). Might still not be a good name, but we can choose one now.

Client code shouldn't directly access that property to avoid breaking changes. We need an API that returns the "original reference".

I also just saw that MPSNodeReference.serialize is incorrectly implemented as `"$PREFIX:$ref"`. It should be `"$PREFIX:${ref.modelReference?.moduleReference?.moduleId}/${ref.modelReference?.modelId}/${ref.nodeId}"` instead. The reason is, that

Reply:
I called it `#mpsNodeRef#` for now. Maybe we should name it `#originalRef#` since we might have other original node sources in the future.